clients/venice-client/src/main/java/com/linkedin/venice/fastclient/StatsAvroGenericStoreClient.java

```diff
@@ -126,8 +126,9 @@ protected CompletableFuture<VeniceResponseMap<K, V>> streamingBatchGet(
       BatchGetRequestContext<K, V> requestContext,
       Set<K> keys) {
     long startTimeInNS = System.nanoTime();
-    CompletableFuture<VeniceResponseMap<K, V>> innerFuture = super.streamingBatchGet(requestContext, keys);
-    return recordMetrics(requestContext, keys.size(), innerFuture, startTimeInNS, clientStatsForBatchGet);
+    CompletableFuture<VeniceResponseMap<K, V>> streamingBatchGetFuture = super.streamingBatchGet(requestContext, keys);
+    recordMetrics(requestContext, keys.size(), streamingBatchGetFuture, startTimeInNS, clientStatsForBatchGet);
```

Comment:
Not an issue, just a question. If we pass the `VeniceResponseCompletableFuture` as the innerFuture for `recordMetrics`. Does it mean in the case of a partial response scenario, metrics are reported when the future times out?

Reply:
Yes actually.
If `streamingBatchGetFuture.get(userTimeout)` times out before all routes results are back but if all results are back within `routingLeakedRequestCleanupThresholdMS` which is more than `userTimeout`, metrics will reflect full success.

For `batchGet(keys)`, even though there is no partial response, the metrics will reflect that there was partial response.

Both are inconsistent with what the end user sees. Will have to check how to control this.

Reply:
As discussed offline, for now, even if future.get(timeout) timeouts, metrics will get incremented when all routers reply back (with proper response or exception) or when `routingLeakedRequestCleanupThresholdMS` times out.